WordPress/Classes/Utility/BackgroundTasks/WeeklyRoundupBackgroundTask.swift

```diff
@@ -34,85 +36,142 @@ private class WeeklyRoundupDataProvider {
         self.onError = onError
     }
 
+    // MARK: API
+
+    /// Fetches the top site statistics from all available sites in the provided context.
+    ///
+    /// This method retrieves all sites in the given context, then fetches the weekly stats
+    /// for each site. The result, which includes the top 5 sites based on the received stats,
+    /// is returned through the provided completion handler.
+    ///
+    /// **Note:** The completion handler is executed on a background queue.
```

Comment:
💡 I know this can be bigger than the scope of this PR, but I think we should avoid leaking background threads outside of their corresponding classes. Adding a note in the documentation is nice for sure, but since we're not passing back managed objects, then it would be better to default to communicating on the main thread.

Reply:
Resolved in https://github.com/wordpress-mobile/WordPress-iOS/pull/21076/commits/8d98c2b2a3f1d31f37df81179d2c1c65bc786109